### DISCUSSION TITLE
Custom fields

### DISCUSSION THREAD

**Discussion by mohhsen67** [ckan/ckan#7716] [General]
I can only define three custom field for datasets. How is it possible to have more custom fields?

**Comment by evilgeniusdev** [comment]
We decided to use the Custom Schema plugin to solve this problem.

https://github.com/ckan/ckanext-scheming

This allowed us to redefine the data collected in its entirety.

**Comment by mohhsen67** [comment]
I installed it using pip install -e "git+https://github.com/ckan/ckanext-scheming.git#egg=ckanext-scheming" but I don't see any new thing in the UI. Should I do anything else?

**Reply by evilgeniusdev** [reply]
You must read the documentation and modify the plugins in the configuration. You will also need to prepare a custom schema for the application to use.

**Comment by mohhsen67** [comment]
I have added the scheming plugin to the plugins like this:

ckan.plugins = activity scheming_datasets

I have also set the following:

scheming.dataset_schemas = ckanext.scheming:subfields.yaml
scheming.presets = ckanext.scheming:presets.json
scheming.dataset_fallback = false

But I still don't see any change in the UI of defining datasets in CKAN.

**Comment by evilgeniusdev** [comment]
You have to custom write a schema as the documentation says. I would suggest you look for a vendor with experience in this to help.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by mohhsen67** [comment]
By writing a custom schema, do you mean, for example, the "subfields.yaml" file which is already existing in ckanext-scheming extension?
